docs/source/migration_guide/2.x_to_3.0.rst

```diff
@@ -0,0 +1,192 @@
+#########################
+Migration from 2.x to 3.0
+#########################
+
+Introduction
+------------
+
+This is a guide to highlight breaking changes in the API and to help the migration of projects from versions 2.X to version 3.0 of the Arduino ESP32 core. 
```

Comment:
Can we mentioned here that v2.X is based on ESP-IDF v4.4 and 3.0 is based on ESP-IDF 5.X?